src/decisionengine_modules/NERSC/sources/NerscSFApi.py

```diff
@@ -0,0 +1,146 @@
+        self.keys_list = ["hours_given", "hours_used", "id", "project_hours_given", "project_hours_used", "repo_name"]
+
+    def check_accesstoken(self, nersc_user):
+        token_url = "https://oidc.nersc.gov/c2id/token"
+        #        currenttime = time.time()
+        renew_bool = False
+
+        rawfile_ucms = "/tmp/ucms_access.token"
+        rawfile_fife = "/tmp/fife_access.token"
+
+        # These 4 variables are defined in /etc/decisionengine/config.d/Nersc.jsonnet
+        #          "ucms_private_key": "/etc/decisionengine/modules.data/ucms_privkey.pem",
+        #          "fife_private_key": "/etc/decisionengine/modules.data/fife_privkey.pem",
+        #          "ucms_client_id":   "/etc/decisionengine/modules.data/ucms_clientid.txt",
+        #          "fife_client_id":   "/etc/decisionengine/modules.data/fife_clientid.txt"
+        # And the actual key files and id files reside in /etc/decisionengine/modules.data/
+        # So, puppet should copy these 4 files from
+        # ssiadmin4:/srv/secrets/per_host/hepcsvc03.fnal.gov/credentials/
+        pemfile_ucms = self.constraints.get("ucms_private_key")
+        pemfile_fife = self.constraints.get("fife_private_key")
+        client_id_ucms = self.constraints.get("ucms_client_id")
+        client_id_fife = self.constraints.get("fife_client_id")
+
+        rawfile = None
+        pemfile = None
+        client_id = None
+        if nersc_user == "uscms":
+            rawfile = rawfile_ucms
+            pemfile = pemfile_ucms
+            with open(client_id_ucms) as ifile:
+                client_id = ifile.read()
+                client_id = client_id.rstrip()
+        elif nersc_user == "fife":
+            rawfile = rawfile_fife
+            pemfile = pemfile_fife
+            with open(client_id_fife) as ifile:
+                client_id = ifile.read()
+                client_id = client_id.rstrip()
+        else:
+            self.logger.error("Unknown user, exiting")
+            return None
+
+        if not os.path.exists(rawfile):
+            self.logger.debug(f"{rawfile} does not exist. Need to generate")
+            renew_bool = True
+        else:
+            atoken = None
+            with open(rawfile) as afile:
+                atoken = afile.read()
+                atoken = atoken.rstrip()
+            # HK> If the access token is expired, the flow goes directly to except jwt.ExpiredSignatureError
+            try:
+                #                result = jwt.decode(atoken, options={"verify_signature": False})
+                jwt.decode(atoken, options={"verify_signature": False})
+                self.logger.debug("Not expired. Returning without generating a new access token")
+                return atoken  # This means the existing access token is not expired.
+
+            except jwt.ExpiredSignatureError:
+                self.logger.debug("Access Token expired")
+                renew_bool = True
+
+        if renew_bool:
```

Comment:
You may not need renew_bool:
```
        if not os.path.exists(rawfile):
            self.logger.debug(f"{rawfile} does not exist. Need to generate")
        else:
            atoken = None
            with open(rawfile) as afile:
                atoken = afile.read()
                atoken = atoken.rstrip()
            # HK> If the access token is expired, the flow goes directly to except jwt.ExpiredSignatureError
            try:
                #                result = jwt.decode(atoken, options={"verify_signature": False})
                jwt.decode(atoken, options={"verify_signature": False})
                self.logger.debug(f"Not expired. Returning {rawfile} without generating a new access token")
                return atoken  # This means the existing access token is not expired.

            except jwt.ExpiredSignatureError:
                self.logger.debug(f"Access Token {rawfile} expired")
```
If the token is present there is the return. The code will arrive here after this block only if it is missing or expired (renew_bool == True)

Reply:
Good catch! I disabled the use of this boolean variable.